src/core/utils/validate-ipni-advertisement.ts

```diff
@@ -107,9 +232,9 @@ export async function validateIPNIAdvertisement(
         await new Promise((resolve) => setTimeout(resolve, delayMs))
         await check()
       } else {
-        // Max attempts reached - don't emit 'complete' event, just throw
-        // The outer catch handler will emit 'failed' event
-        const msg = `IPFS root CID "${ipfsRootCid.toString()}" not announced to IPNI after ${maxAttempts} attempt${maxAttempts === 1 ? '' : 's'}`
+        // Max attempts reached - validation failed
+        const msgBase = `IPFS root CID "${ipfsRootCid.toString()}" not announced to IPNI after ${maxAttempts} attempt${maxAttempts === 1 ? '' : 's'}`
+        const msg = lastFailureReason != null ? `${msgBase}. Last observation: ${lastFailureReason}` : msgBase
```

Comment:
It would be nice if the error message included the expectedMultiaddrs and what the response actually had in case we have some unexpected issue (e.g., one of them is adding the port while the other isn't).  This will make it easier to debug.